c++/src/kj/mutex.c++

```diff
@@ -547,8 +547,16 @@ void Mutex::wait(Predicate& predicate, Maybe<Duration> timeout) {
     } else {
       DWORD error = GetLastError();
       if (error == ERROR_TIMEOUT) {
-        // Timed out. Skip predicate check.
-        return;
+        // As mentioned above, Windows sleeps are inaccurate, so we must double check.
+        KJ_IF_MAYBE(e, endTime) {
+          if (*e <= clock->now()) {
+            // Actually timed out. Skip predicate check.
+            return;
+          }
```

Comment:
For the case where we woke up too early, maybe comment to explain that we expect the predicate is still false, but we're going to proceed around the loop anyway just for the sake of setting up a new sleep.

Reply:
And done.